woocommerce/woocommerce-gutenberg-products-block
# Store Customization > Implement Client for requests to the GPT API

## What

Create a new client for handling the POST requests to the GPT API so it can return us the IDs of the verticals that are a match for a given store.

In this first moment, to expedite work, this util will live within WooCommerce Blocks. A separate issue will be created to move it to the core of WooCommerce, so other teams/projects working with AI can use the same client for their requests.

PS: The class for configuring JetPack without requiring the plugin to be installed is also introduced in this PR, but we won't rely on it for the MVP within WooExpress, as all sites are already connected to JetPack and WP.com in that environment.

🎯 #10617

## Why


This is a requirement to ensure all of our patterns can have AI-managed images and, later on, AI-generated content.

## Testing Instructions



_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Create a new [JN install](https://jurassic.ninja/) with both Jetpack and Woocommerce installed and activated.
3. Install and activate WooCommerce Blocks from this zip file: [woocommerce-gutenberg-products-block.zip](https://github.com/woocommerce/woocommerce-blocks/files/12578680/woocommerce-gutenberg-products-block.zip)

5. In your wp-admin, you should see the following section for Jetpack:

<img width="1803" alt="Screenshot 2023-09-11 at 18 55 38" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/fc804759-f724-4fa2-829d-7bfd441962ec">

6. Click on "Set up Jetpack" and make sure the connection is successful:

<img width="1994" alt="Screenshot 2023-09-11 at 18 58 30" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/661ee4c5-a0ac-40fd-9e5e-835a0e4c1cb7">

7. Head over to Tools > Plugin editor:

<img width="337" alt="Screenshot 2023-09-11 at 18 34 23" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/c353bd13-bf00-4bc8-93d6-d827ca99997f">

8. Select the WooCommerce Blocks plugin and add the following within your `woocommerce-blocks/woocommerce-gutenberg-products-block.php` file: 

```php
function register_and_connect() {
	update_option( 'woocommerce_blocks_allow_ai_connection', true );

	$ai_connection = new \Automattic\WooCommerce\Blocks\AI\Connection();
	$site_id       = $ai_connection->get_site_id();
	$token         = $ai_connection->get_jwt_token( $site_id );

	$ai_response = $ai_connection->fetch_ai_response( $token, 'Tell me a joke about ducks.' );

	error_log( print_r( $ai_response, true ) );
}

add_action( 'admin_init', 'register_and_connect' );
```

9. Now access your JN install via SSH (you can copy the command for access over here:)

<img width="346" alt="Screenshot 2023-09-11 at 18 38 44" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/419f338f-d203-4bec-8f78-4589d5ad3ad2">

10. Head over to your wp-content directory and tail the debug.log file. Example:

```
cd /srv/users/user964902c9/apps/user964902c9/public/wp-content
tail -f debug.log
```
11. Now go back to your browser and refresh the page in the main admin page e.g. `https://your-site.jurassic.ninja/wp-admin`

12. Go back to your terminal and confirm that you can see the AI output for your prompt:

<img width="883" alt="Screenshot 2023-09-11 at 18 45 34" src="https://github.com/woocommerce/woocommerce-blocks/assets/15730971/6afab334-a311-4d3e-a460-870a85943142">


## Screenshots or screencast



Not applicable.

## WooCommerce Visibility


Required:


## Checklist

Required:

Conditional:

## Changelog


N/A
